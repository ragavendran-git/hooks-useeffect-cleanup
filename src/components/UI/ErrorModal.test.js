import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from "enzyme-to-json";
import React from 'react';

import ErrorModal from './ErrorModal';

configure({ adapter: new Adapter() });

it("Should render correctly", () => {
    const wrapper = shallow(<ErrorModal />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it("Should call onCloseHandler function", () => {
    const onCloseHandler = jest.fn();
    const wrapper = mount(<ErrorModal onClose={onCloseHandler}/>);

    wrapper.find("button").simulate("click");
    expect(onCloseHandler).toHaveBeenCalled();
});