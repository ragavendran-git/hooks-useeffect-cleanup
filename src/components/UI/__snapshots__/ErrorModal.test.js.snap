// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<Fragment>
  <div
    className="backdrop"
  />
  <div
    className="error-modal"
  >
    <h2>
      An Error Occurred!
    </h2>
    <p />
    <div
      className="error-modal__actions"
    >
      <button
        type="button"
      >
        Okay
      </button>
    </div>
  </div>
</Fragment>
`;
